src/helper/tools/oval-tool.js

```diff
@@ -89,7 +89,21 @@ class OvalTool extends paper.Tool {
         const downPoint = new paper.Point(event.downPoint.x, event.downPoint.y);
         const point = new paper.Point(event.point.x, event.point.y);
         if (event.modifiers.shift) {
-            this.oval.size = new paper.Point(event.downPoint.x - event.point.x, event.downPoint.x - event.point.x);
+            // These variables are used for determining the relative quadrant that the circle will appear in.
+            // So if you drag up and right, it'll show up above and to the right of where you started dragging, etc.
+            let offsetX = event.point.x - event.downPoint.x;
+            let offsetY = event.point.y - event.downPoint.y;
+
+            // If the offset variables are zero, the circle ends up having zero width or height, which is bad.
```

Comment:
This step isn't necessary because if you look at line 128, if you draw an oval that's too small it will just get removed.

Reply:
I think the problem was that if you drag a circle out to, for example, X = 100 and Y = 0 (relative to where you started dragging), you would *expect* to see a 100-unit circle, but there is none (because offsetX/Y = 0). The actual case where this is jarring is when you go from Y = 1 to Y = 0.